onmt/inputters/inputter.py

```diff
@@ -246,6 +246,24 @@ def _build_field_vocab(field, counter, **kwargs):
     field.vocab = field.vocab_cls(counter, specials=specials, **kwargs)
 
 
+def _load_vocab(vocab_path, name, counters):
+    # counters changes in place
+    vocab = _read_vocab_file(vocab_path, name)
+    vocab_size = len(vocab)
+    logger.info('Loaded %s vocab has %d tokens.' % (name, vocab_size))
+    for i, token in enumerate(vocab):
+        # keep the order of tokens specified in the vocab file by
+        # adding them to the counter with decreasing counting values
+        counters[name][token] = vocab_size - i
+    return vocab, vocab_size
+
+
+def _build_fv_from_multifield(multifield, counters, build_fv_args):
+    for name, field in multifield:
+        _build_field_vocab(field, counters[name], **build_fv_args[name])
+        logger.info(" * %s vocab size: %d." % (name, len(field.vocab)))
+
```

Comment:
Creating these 2 functions may trigger a future issue in checkpoints if we change again.
Could we make them local in build_vocab ?

Reply:
Sure although I honestly don't see how that would come up. Neither those functions nor ``build_vocab`` should be checkpointed, right?

Reply:
no, you're correct. But now I am confused why _feature_tokenize WAS checkpointed in the version between 14 days ago and this morning commit.

Reply:
It still is. [``text_fields``](https://github.com/OpenNMT/OpenNMT-py/blob/master/onmt/inputters/text_dataset.py#L90) - used to be part of ``get_fields`` - passes a partial of ``_feature_tokenize`` into ``torchtext.data.Field``. [``Field``](https://github.com/pytorch/text/blob/master/torchtext/data/field.py#L163) makes it as an attr. And ``model_saver.py`` [``ModelSaver``](https://github.com/OpenNMT/OpenNMT-py/blob/f4036abded5a5494aadc7c6aa9366f7c40c17894/onmt/models/model_saver.py#L110) pickles the fields.